docs/source/en/api/pipelines/stable_diffusion/adapter.mdx

```diff
@@ -0,0 +1,194 @@
+
+The abstract of the paper is the following:
+
+*The incredible generative ability of large-scale text-to-image (T2I) models has demonstrated strong power of learning complex structures and meaningful semantics. However, relying solely on text prompts cannot fully take advantage of the knowledge learned by the model, especially when flexible and accurate structure control is needed. In this paper, we aim to ``dig out" the capabilities that T2I models have implicitly learned, and then explicitly use them to control the generation more granularly. Specifically, we propose to learn simple and small T2I-Adapters to align internal knowledge in T2I models with external control signals, while freezing the original large T2I models. In this way, we can train various adapters according to different conditions, and achieve rich control and editing effects. Further, the proposed T2I-Adapters have attractive properties of practical value, such as composability and generalization ability. Extensive experiments demonstrate that our T2I-Adapter has promising generation quality and a wide range of applications.*
+
+## Available Pipelines:
+
+| Pipeline | Tasks | Demo
+|---|---|:---:|
+| [StableDiffusionAdapterPipeline](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_adapter.py) | *Text-to-Image Generation with T2I-Adapter Conditioning* | -
+
+## Usage example
+
+In the following we give a simple example of how to use a *Adapter* checkpoint with Diffusers for inference.
+The inference pipeline is the same for all pipelines:
+
+ 1. Take an image and run it through a pre-conditioning processor to obtain *control image*.
+ 2. Run the pre-processed *control image* and *prompt* through the [`StableDiffusionAdapterPipeline`].
+
+Let's have a look at a simple example using the [Color Adapter](https://huggingface.co/RzZ/sd-v1-4-adapter-color).
+
+```python
+from diffusers.utils import load_image
+
+image = load_image("https://huggingface.co/RzZ/sd-v1-4-adapter-color/resolve/main/color_ref.png")
+```
+
+![img](https://huggingface.co/RzZ/sd-v1-4-adapter-color/resolve/main/color_ref.png)
+
+
+Then we can create our color palette by simply resize it to 8 by 8 pixels then scale it back to original size.
+
+```python
+from PIL import Image
+
+color_palette = image.resize((8, 8))
+color_palette = color_palette.resize((512, 512), resample=Image.Resampling.NEAREST)
+```
+
+Let's take a look at the processed image.
+
+![img](https://huggingface.co/RzZ/sd-v1-4-adapter-color/resolve/main/color_palette.png)
+
+
+After we having `color_palette` in hand, we can create the [`StableDiffusionAdapterPipeline`] with pretrained checkpoint.
+
+```py
+import torch
+from diffusers import StableDiffusionAdapterPipeline, Adapter
+
+adapter = Adapter.from_pretrained("RzZ/sd-v1-4-adapter-color")
+pipe = StableDiffusionAdapterPipeline.from_pretrained(
+    "RzZ/sd-v1-4-adapter",
```

Comment:
Doesn't the model use pre-trained stable diffusion? We shouldn't have to load from a custom pipeline

Reply:
*CompVis/stable-diffusion-v1-4* will also work here, *RzZ/sd-v1-4-adapter* is created for testing download entire pipeline with `StableDiffusionAdapterPipeline.from_pretrained`  without the `adapter` argument.

Reply:
While technically the pipeline will allow downloading all at once, we shouldn't ever document that as it might make users think it's the recommended api. Could you go through all examples where we load the pipeline and make it point at the official stable diffusion checkpoint? Thanks :)

Reply:
Updated with **CompVis/stable-diffusion-v1-4**